src/rust/iced-x86-py/src/instruction.rs

```diff
@@ -166,6 +169,31 @@ impl Instruction {
 		Self { instr: iced_x86::Instruction::default() }
 	}
 
+	/// Set the internal state with the given unpickled state.
+	///
+	/// Args:
+	///     state (Any): unpickled state
+	#[pyo3(text_signature = "($self, state, /)")]
+	fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+		match state.extract::<&PyBytes>(py) {
+			Ok(s) => {
+				self.instr = deserialize(s.as_bytes()).unwrap();
```

Comment:
Return an error instead of panicing here (`unwrap()`). Other funcs in the same file map errors to `to_value_error`, I think that's an OK exception to generate if the input is invalid.